apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PatientResourceProvider.java

```diff
@@ -97,13 +100,13 @@ public Patient read(@IdParam IdType patientId, RequestDetails requestDetails) {
     String beneIdText = patientId.getIdPart();
     if (beneIdText == null || beneIdText.trim().isEmpty()) throw new IllegalArgumentException();
 
-    IncludeIdentifiersMode includeIdentifiersMode =
-        IncludeIdentifiersMode.determineIncludeIdentifiersMode(requestDetails);
+    List<String> includeIdentifiersValues = returnIncludeIdentifiersValues(requestDetails);
 
     CriteriaBuilder builder = entityManager.getCriteriaBuilder();
     CriteriaQuery<Beneficiary> criteria = builder.createQuery(Beneficiary.class);
     Root<Beneficiary> root = criteria.from(Beneficiary.class);
-    if (includeIdentifiersMode == IncludeIdentifiersMode.INCLUDE_HICNS_AND_MBIS) {
+
+    if (hasHICN(includeIdentifiersValues) || hasMBI(includeIdentifiersValues)) {
```

Comment:
split this up. `if has hicn; load hicn history` `if has mbi; load mbi history`

Reply:
TY! I split this up.